pkg/ext-proc/backend/datastore.go

```diff
@@ -59,13 +112,58 @@ func (ds *K8sDatastore) getInferencePool() (*v1alpha1.InferencePool, error) {
 	return ds.inferencePool, nil
 }
 
-func (ds *K8sDatastore) GetPodIPs() []string {
-	var ips []string
-	ds.pods.Range(func(name, pod any) bool {
-		ips = append(ips, pod.(*corev1.Pod).Status.PodIP)
-		return true
-	})
-	return ips
+func (ds *K8sDatastore) createPodLister(pool *v1alpha1.InferencePool) *PodLister {
```

Comment:
The only reasonable tests that we can add to test this are integration tests; but the integration tests setup we have right now is not complete and needs more work. However, the existing e2e test offers some coverage, in addition I tested manually several cases including scaling down pods and scaling up pods, changing inferencePool selector.